src/handshakestate.rs

```diff
@@ -284,13 +284,14 @@ impl HandshakeState {
         Ok(payload_len)
     }
 
-    pub fn get_remote_static(&self) -> Option<&[u8; MAXDHLEN]> {
-        self.rs.as_option_ref()
+    pub fn get_remote_static(&self) -> Option<&[u8]> {
+        self.rs.as_option_ref().map(|rs| &rs[..self.dh_len()])
```

Comment:
I modified this function to return a slice, as it wasn't being used anywhere else, and it meant we didn't need to expose `dh_len` inside `Session`.